lib/actions-oclif/os/configure.ts

```diff
@@ -0,0 +1,430 @@
+
+		if (options.device) {
+			device = await balena.models['device'].get(options.device);
+			deviceTypeSlug = device.device_type;
+		} else {
+			app = await balena.models['application'].get(options.application!);
+			await checkDeviceTypeCompatibility(balena, options, app);
+			deviceTypeSlug = options['device-type'] || app.device_type;
+		}
+
+		const deviceTypeManifest = await helpers.getManifest(
+			params.image,
+			deviceTypeSlug,
+		);
+
+		let configJson: import('../../utils/config').ImgConfig | undefined;
+		if (options.config) {
+			const rawConfig = await fs.readFile(options.config, 'utf8');
+			configJson = JSON.parse(rawConfig);
+		}
+
+		const answers: Answers = await askQuestionsForDeviceType(
+			deviceTypeManifest,
+			options,
+			configJson,
+		);
+		if (options.application) {
+			answers.deviceType = deviceTypeSlug;
+		}
+		answers.version =
+			options.version ||
+			(await getOsVersionFromImage(params.image, deviceTypeManifest, devInit));
+
+		if (_.isEmpty(configJson)) {
+			if (device) {
+				configJson = await generateDeviceConfig(
+					device as DeferredDevice,
+					options['device-api-key'],
+					answers,
+				);
+			} else {
+				configJson = await generateApplicationConfig(app!, answers);
+			}
+		}
+
+		console.info('Configuring operating system image');
+
+		await helpers.osProgressHandler(
+			await devInit.configure(
+				params.image,
+				deviceTypeManifest,
+				configJson || {},
+				answers,
+			),
+		);
+	}
+}
+
+async function validateOptions(options: FlagsDef) {
+	if (process.platform === 'win32') {
+		throw new ExpectedError(stripIndent`
+			Unsupported platform error: the 'balena os configure' command currently requires
+			the Windows Subsystem for Linux in order to run on Windows. It was tested with
+			the Ubuntu 18.04 distribution from the Microsoft Store. With WSL, a balena CLI
+			release for Linux (rather than Windows) should be installed: for example, the
+			standalone zip package for Linux. (It is possible to have both a Windows CLI
+			release and a Linux CLI release installed simultaneously.) For more information
+			on WSL and the balena CLI installation options, please check:
+			- https://docs.microsoft.com/en-us/windows/wsl/about
+			- https://github.com/balena-io/balena-cli/blob/master/INSTALL.md
+		`);
+	}
+	// The 'device' and 'application' options are declared "exclusive" in the oclif
+	// flag definitions above, so oclif will enforce that they are not both used together.
+	if (!options.device && !options.application) {
+		throw new ExpectedError(
+			"Either the '--device' or the '--application' option must be provided",
+		);
+	}
+	if (!options.application && options['device-type']) {
+		throw new ExpectedError(
+			"The '--device-type' option can only be used in conjunction with the '--application' option",
+		);
+	}
+	if (options['device-api-key']) {
+		console.error(stripIndent`
+			-------------------------------------------------------------------------------------------
```

Comment:
Intended NL?

Reply:
Not sure I understand the question. At run time, what gets printed by the CLI is:
```
$ ./bin/balena-dev os configure --device-api-key foo -a test-rpi os.img
-------------------------------------------------------------------------------------------
Warning: The --device-api-key option is deprecated and will be removed in a future release.
A suitable key is automatically generated or fetched if this option is omitted.
-------------------------------------------------------------------------------------------
? Network Connection (Use arrow keys)
❯ ethernet
  wifi
...
```